### ISSUE TITLE
controllers loaded but the robots falls 

### ISSUE THREAD

**Issue by peakyquest** [ros-controls/gz_ros2_control#426]
I am using humble with gazebo ignition. I am able to spawn the robot in the ignition env, and I all the controllers are active, still my robot falls on the ground, and when I am publishing something on the topic the robot doesnt move

**Comment by christophfroehlich** [comment]
Please give more information to be able to help you.

Which controllers are you using? 
For example, add output of `ros2 control list_controllers` and `ros2 control list_hardware_components`

**Comment by peakyquest** [comment]
![IMG_5257](https://github.com/user-attachments/assets/757b1cca-ba7a-4ad3-8a5d-d824c7d224f7)
![IMG_5256](https://github.com/user-attachments/assets/78873fea-7da5-4329-b880-e2caaaf4d6dc)
![IMG_5259](https://github.com/user-attachments/assets/4a85779e-86e2-4f89-b9fb-b4b8460c471c)


I have added damping and fiction in my urdf, which makes my robot stand, but I am not sure is this a correct way. Further more I was unable to move the robot with the trajectory topics, should I have to use ign_bridge but there is no ign_bridge used in the ros2 control demos videos. A bit confused.

**Comment by christophfroehlich** [comment]
As you are using joint_trajectory_controller, you have to send topics to the `/<controller_name>/joint_trajectory` topics or use its action interface.

Please don't post screenshots, better copy the log as txt in a code fraction or upload a txt file if it is too long.

Which command interfaces are you using? As I already asked for: `ros2 control list_hardware_components`

**Comment by peakyquest** [comment]
Hardware Component 1
	name: IgnitionSystem
	type: 
	plugin name: 
	state: id=3 label=active
	command interfaces
		joint_1/position [available] [claimed]
		joint_2/position [available] [claimed]
		joint_3/position [available] [claimed]
		joint_4/position [available] [claimed]
		joint_5/position [available] [claimed]
		joint_6/position [available] [claimed]

**Comment by christophfroehlich** [comment]
Have you tried to send something to the controllers with the mentioned topics, e.g., via rqt_joint_trajectory_controller?
Falling robots could also be a result from wrong masses/inertias of the links -> the solver explodes.

**Comment by peakyquest** [comment]
[urdf.txt](https://github.com/user-attachments/files/17393165/urdf.txt)
[controller.txt](https://github.com/user-attachments/files/17393166/controller.txt)
[launch.txt](https://github.com/user-attachments/files/17393168/launch.txt)

well I use rqt_joint_trajectory_controller but the robot doesnt move either. 
and I have use mu old urdf, which worked fine on gazebo classic, so I am sure that it isnt about masses and inertia, however this was the link which I get my urdf frrom
https://github.com/ros-industrial/kuka_experimental/blob/melodic-devel/kuka_kr210_support/urdf/kr210l150.urdf

**Comment by kdeshp-prof** [comment]
Is there any resolution on this one!! I seem to have same problem with my custom mobile manipulator
ROS2 Jazzy, GZ Harmonic

**Comment by christophfroehlich** [comment]
@kdeshp-prof you opened the other issue #474 right?
@peakyquest have you fixed your problem? If you want us to introspect your setup, then you have to provide a full example / package to reproduce it.

**Comment by kdeshp-prof** [comment]
@christophfroehlich #474 was not opened by me, the solution mentioned by you their works.
But, my problem with my custom robot is still unresolved.
The controllers become active for my manipulator and the manipulator is still falling

After controllers becoming active this is the only warning or debug message I see.
[gazebo-1] [WARN] [1729980256.923015232] [gz_ros_control]:  Desired controller update period (0.01 s) is slower than the gazebo simulation period (0.001 s).
[gazebo-1] [Dbg] [SDFFeatures.cc:332] Mesh construction from an SDF has not been implemented yet for dartsim. Use AttachMeshShapeFeature to use mesh shapes.
[gazebo-1] [Dbg] [SDFFeatures.cc:863] The geometry element of collision [joint] couldn't be created

Since, the custom robot belongs to a proprietary vendor making the URDF public or other piece of code is not possible
From, the above can you please tell me if I need to fix something in the URDF or the problem lies on control side?

**Comment by christophfroehlich** [comment]
Can you try a different physics engine than dart? We don't have tests for that, it may be that something is working different there.

**Comment by peakyquest** [comment]
yes, I have changed the model, I was first working on kuka robotic arm, but the urdf has some issues, the inertia of the links were not correct, so I moved to the panda robotic arm, the ros2 controller work properly on the panda's sdf model.

**Comment by peakyquest** [comment]
@kdeshp-prof I will suggest you try to configure the proper inertia of the robot, or try to find robot similar to yours on the following website, an then try to add ros2_controller [Fuel Ignition](https://app.gazebosim.org/fuel/models)

**Comment by christophfroehlich** [comment]
Closing because the problem seems to be solved on the URDF side.
